tests/components/side-panel/side-panel-header.spec.tsx

```diff
@@ -0,0 +1,63 @@
+import { render, screen } from "@testing-library/react";
+import "jest-canvas-mock";
+import { SidePanelHeader } from "src/components/side-panel/side-panel-header";
+import { EBuilderMode, EElementType } from "src/context-providers";
+import { ELEMENT_BUTTON_LABELS } from "src/data/elements-data";
+import { TestHelper } from "src/util/test-helper";
+
+describe("side-panel-header", () => {
+    afterEach(() => {
+        jest.restoreAllMocks();
+        jest.resetAllMocks();
+    });
+
+    describe("getHeaderTitle", () => {
+        it("should run the getHeaderTitle function and display the Add Elements title for the add-elements mode", () => {
+            renderComponent();
+            expect(getHeaderLabel().textContent).toEqual("Add elements");
+        });
+
+        it("should run the getHeaderTitle function and display the Add Pages title for the add-pages mode", () => {
+            renderComponent({
+                builderContext: {
+                    mode: EBuilderMode.EDIT_PAGES,
+                },
+            });
+            expect(getHeaderLabel().textContent).toEqual("Add pages");
+        });
+
+        it("should display the edit element header title when there is a focusedElement", () => {
+            renderComponent({
+                builderContext: {
+                    focusedElement: {
+                        element: mockElement,
+                    },
+                },
+            });
+            expect(getHeaderLabel().textContent).toEqual("Edit details");
+        });
+    });
+});
```

Comment:
add test for the conditional rendering of the cross icon

Reply:
![image](https://github.com/LifeSG/web-form-builder/assets/103546625/9db44282-9f52-46fb-bdda-89ec5ca528d2)

I created the test alrdy, its been pushed and it passes as well :)